modules/sfp_ivre.py

```diff
@@ -0,0 +1,342 @@
+# -*- coding: utf-8 -*-
+# -------------------------------------------------------------------------------
+# Name:         sfp_ivre
+# Purpose:      Query an IVRE instance
+#
+# Author:      Pierre Lalet <pierre@droids-corp.org>
+#
+# Created:     2021-09-14
+# Copyright:   (c) Pierre Lalet
+# Licence:     GPL
+# -------------------------------------------------------------------------------
+
+import hashlib
+
+from netaddr import IPNetwork
+from ivre.db import db, DBPassive
+from ivre import utils
+
+from spiderfoot import SpiderFootEvent, SpiderFootPlugin
+
+
+def flt_from_asnum(dbase, asnum):
+    return dbase.searchasnum(int(asnum))
+
+
+def flt_from_ip(dbase, addr):
+    yield dbase.searchhost(addr)
+
+
+def flt_from_net(dbase, net):
+    yield dbase.searchnet(net)
+
+
+def flt_from_port(dbase, proto, port):
+    try:
+        yield dbase.searchport(port, protocol=proto)
+    except ValueError:
+        yield dbase.searchnonexistent()
+
+
+def flt_from_cert(dbase, cert):
+    yield dbase.searchcert(
+        sha256=hashlib.new("sha256", utils.decode_b64(cert.encode())).hexdigest()
+    )
+
+
+def flt_from_fqdn(dbase, name):
+    if isinstance(dbase, DBPassive):
+        yield dbase.searchdns(name=name)
+        yield dbase.searchdns(name=name, reverse=True)
+    else:
+        yield dbase.searchhostname(name)
+
+
+EVENTS_FILTERS = {
```

Comment:
Any particular reason why this approach was taken? It's quite different to all the other modules.